src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,18 +28,30 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        String[] splittedAddress = trimmedAddress.split(",");
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(splittedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        } else {
+            block = new Block(splittedAddress[0].trim());
+            street = new Street(splittedAddress[1].trim());
+            unit = new Unit(splittedAddress[2].trim());
+            postalCode = new PostalCode(splittedAddress[3].trim());
+            
         }
         this.value = trimmedAddress;
```

Comment:
there is no need to store a duplicate of the address since you can reconstruct from the component classes